tests/testthat/test-prune.R

```diff
@@ -0,0 +1,20 @@
+library(yspec)
+library(testthat)
+
+context("text-prune")
+
+test_that("ys_prune selects available columns", {
+  data <- ys_help$data()
+  spec <- ys_help$spec()
+  data$STUDY <- NULL
+  data$TAD <- NULL
+  ans <- ys_prune(data, spec)
```

Comment:
@kylebaron 

should this test have a data frame with more columns than the spec?

For example

- data has columns: A, B, C
- spec has columns: A, B

The current test has the reverse, and therefore:

`names(ys_prune(data, spec)) == names(data)`

Reply:
Yes; I was intending fewer columns. Can add more columns too.

Reply:
Added two columns matching nothing.

Reply:
@kylebaron 

One other question - if I add:

`data <- data[, rev(names(data)), drop = FALSE]`

after line 12 (`data$BAR <- 2`) but before line 13 (`ans <- ys_prune(data, spec))`), the test then fails.

Is this OK?

Reply:
Thanks, @andersone1 ; yes, there was something wrong in the logic for matching / ordering the names.  

- Fixed the logic in `ys_prune`
- Randomized the names in `data` in the test
- Added expectations for reporting columns not there

Thanks!

Reply:
I reverted that one expectation; I do think we want this to be the check:

```r
  expect_equal(names(ans), names(spec2))
```

The names in the final answer are identical (including order) to the names in the spec, minus the columns that we dropped.